code/API_definitions/device-status.yaml

```diff
@@ -389,71 +466,75 @@ components:
-      enum:
-        - ROAMING_STATUS
-    DeviceStatusEventTypeList:
-      description: List of Device Status Event Type
-      type: array
-      items:
-        $ref: '#/components/schemas/DeviceStatusEventType'
-      example: [ ROAMING_STATUS]
-    CreateSubscription:
-      description: The request for creating a Device Status Subscription
+        
+    CreateEventSubscription:
+      description: The request for creating a Device Status event subscription
       type: object
+      required:
+        - webhook
       properties:
-        notificationUrl:
-          type: string
-          example: "https://application-server.com"
-          description: Allows asynchronous delivery of device status related events
-        notificationAuthToken:
-          type: string
-          example: "c8974e592c2fa383d4a3960714"
-          description: Authentification token for callback API
         subscriptionDetail:
-          $ref: '#/components/schemas/SubscriptionDetail'
+          $ref: '#/components/schemas/EventSubscriptionDetail'
         subscriptionExpireTime:
           type: string
           format: date-time
           example: 2023-01-17T13:18:23.682Z
           description: The subscription expiration time in date-time format.
+        webhook:
+          $ref: '#/components/schemas/Webhook'
+
+    Webhook:
+      description: Webhook information for event channel
+      type: object
       required:
-        - ueId
         - notificationUrl
-    SubscriptionDetail:
-      description: The detail of the requested description
+      properties:
+        notificationUrl:
+          type: string
+          example: "https://application-server.com"
+          description: https callback address where the event notification must be POST-ed
+        notificationAuthToken:
+          type: string
+          example: "c8974e592c2fa383d4a3960714"
+          description: | 
+            OAuth2 token to be used by the callback API endpoint. It MUST be indicated within HTTP Authorization header e.g. Authorization: Bearer $notificationAuthToken
+
+    EventSubscriptionDetail:
+      description: The detail of the requested event subscription
       type: object
+      required:
+        - eventType
       properties:
-        ueId:
-          $ref: '#/components/schemas/UeId'
-        uePort:
-          $ref: '#/components/schemas/Port'
-    CreateRoamingSubscription:
-      allOf:
-        - $ref: '#/components/schemas/CreateSubscription'
-        - type: object
-    UeIdList:
-      description: List of User equipment identifiers
-      type: array
-      items:
-        $ref: '#/components/schemas/UeId'
-    SubscriptionInfo:
+        device:
+          $ref: '#/components/schemas/Device'
+        eventType:
+          $ref: '#/components/schemas/RoamingEventType'
+
+    RoamingEventType:
+      type: string
+      description: |
+        ROAMING_STATUS - Event triggered when the device switch from roaming ON to roaming OFF and conversely
```

Comment:
Looking at the event generated by a ROAMING_STATUS event subscription, both ON/OFF as well as country code/name is returned. 

I thereby assume that subscribing to ROAMING_STATUS will give you notifications for on, off as well change country. 

For clarity this should be reflected in the description.

Reply:
If I, with a UK SIM, were in, for example, France and travelled to Germany, my `roaming` state would remain unchanged at `true`. So I would not expect a `ROAMING_STATUS` change event to be generated.

But if this is not clear, then indeed the description should be improved.

Reply:
Good comment Eric. Then I am wondering about the ROAMING_CHANGE_COUNTRY event type. That clearly indicate that I would get a roaming status notification in the situation you describe above to indicate that you move from one country to another while roaming. 

I was thinking of the ROAMING_ON, ROAMING_OFF and ROAMING_CHANGE_COUNTRY as a subset of ROAMING_STATUS. Maybe that is not the case after all?

Reply:
Maybe we are saying the same thing but if I do not misunderstand the 3GPP standard you will get a notification only when changing from HPLMN to VPLMN and vice versa. You will not get a notification when moving from one VPLMN to another VPLMN. Even though there are implementations that will give you also notifications when changing between VPLMNs we cannot rely on it. Thereby the name of the event ROAMING_CHANGE_COUNTRY is a bit misleading as it indicates that you will always get such notifications.

Reply:
Well, I think the use case for `ROAMING_CHANGE_COUNTRY` is clear - the API consumer will get a notification whenever the MCC of the PLMN ID of the network that the device is connected to changes.

As for getting this information, you are correct that the NEF will not generate a ROAMING_STATUS monitoringType event just because the device has changed VPLMN. However, that does not mean that the NEF cannot be used to get this information. The NEF can be regularly asked (i.e. polled) for a one-time report as to the current roaming status and PLMN ID. For this use case, the polling would not have to be very frequent, and would only have to be done for devices for which `ROAMING_CHANGE_COUNTRY` events have been requested.

And, as you mention above, there are proprietary implementations that can give this information. Indeed, we are using a proprietary implementation in Vodafone, and this use case is important to us. If any API provider does not want to implement `ROAMING_CHANGE_COUNTRY` events because the NEF does not directly give these notifications, then they are free to do that.

Reply:
@fernandopradocabrillo
What you are missing is that the different notification types are something that the API consumer subscribes to (i.e. is an input when they create a subscription). So if an API consumer subscribed to `ROAMING_ON`, they would ONLY get a notification when the device moved from the HPLMN to a VPLMN.

To get the notifications in your example, you would need to subscribe to all three of the `ROAMING_ON`, `ROAMING_CHANGE_COUNTRY` and `ROAMING_OFF` events (i.e. have 3 separate subscriptions).

Reply:
Yes, that part understood! My question is, what is the difference between subscribing only to `ROAMING_STATUS` or subscribing to both `ROAMING_ON` and `ROAMING_OFF`? What are the benefits of subscribing to `ROAMING_STATUS` only apart of having to subscribe only to one event? 

In the current model, if I subscribe only to `ROAMING_ON` because I am only interested in that event, I do not receive information on which country is the SIM roaming, I am forced to subscribe also to `ROAMING_STATUS` to get that information.

And If we say that the `ROAMING_STATUS` represents a change in roaming status, why don't we include the country change in the actions that cover this event?

Reply:
The difference is that you only have one subscription to manage instead of two. The only reason to subscribe to both `ROAMING_ON` and `ROAMING_OFF` rather than `ROAMING_STATUS`, is because you need separate subscriptions, possibly with separate notification endpoints and certainly with different `eventSubscriptionId`. Otherwise, use `ROAMING_STATUS`.

There is an argument for adding roaming country for `ROAMING_ON` notifications, but I don't know the use case for those events. It may be that the use case only requires the API consumer to know that the device is now roaming, but it does not matter where. This is not a use case that Vodafone is interested in.

`ROAMING_STATUS` notifications can include the new country that triggered the event. Have a look at the example in the API definition. If a client wants to know what the previous country was, they need to store the previous notification.

Reply:
Just checking , even if we include country information in the ROAMING_ON and ROAMING_OFF events, can't we control which clients are authorized to view the country code information and restrict it for those who don't have the necessary privileges . Sorry , if doesn't make sense.

Reply:
Okey then, fine from my side!